### PR TITLE
[TML/BC/OGS/helpers] prepare for 2D caclculations xy-xz plane transfo…

### PR DESCRIPTION
**Pull request by jbathmann**
…rmation

In case the grid for groundwater flows is provided in the x-y-plane, the cellinformation helper is now able to finde cells associated to trees in the correct way.

### CODE REVIEW

File: TreeModelLib/BelowgroundCompetition/OGS/helpers/CellInformation.py [hunk 1]
```diff
@@ -52,6 +52,9 @@ def getCellIDsAtXY(self, x, y):
             x = 0
         p1 = [x, y, bounds[-1]]
         p2 = [x, y, bounds[-2]]
+        if np.abs(bounds[-1]-bounds[-2])<0.01:
```
Comment (mcwimm) [inline TreeModelLib/BelowgroundCompetition/OGS/helpers/CellInformation.py:55]: Why '0.01' and not 0?

Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGS/helpers/CellInformation.py]: Due to numerical unertainities. 
However, I reduced the \epsilon. Is this ok for you?

Comment (mcwimm) [reply TreeModelLib/BelowgroundCompetition/OGS/helpers/CellInformation.py]: Again, some comments would be great

Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/OGS/helpers/CellInformation.py]: added

File: TreeModelLib/BelowgroundCompetition/OGS/helpers/CellInformation.py [hunk 2]
```diff
@@ -43,15 +43,25 @@ def getCellId(self, x, y, z):
...
+        elif np.abs(bounds[0] - bounds[1]) < 0.0001:
             x = 0
         p1 = [x, y, bounds[-1]]
         p2 = [x, y, bounds[-2]]
+        # TODO: find better solution
+        # If the mesh is in 2D and in the x-y plane, it is probably so as OGS
+        # only processes 2D meshes in the x-y- plane. Hence, a rotation is per-
+        # formed here.
+        if np.abs(bounds[-1] - bounds[-2]) < 0.0001:
+            p1 = [x, bounds[2], bounds[-1]]
```
Comment (mcwimm) [inline TreeModelLib/BelowgroundCompetition/OGS/helpers/CellInformation.py:60]: I'm a bit confused on which cases are distinguished here (as you also overwrite p1 and p2 here and than ln. 48-54 are redundant). 
Suggestion (but not sure if I still struggle to understand it):
`
if np.abs(bounds[-1] - bounds[-2]) < 0.0001:
  p1 =...
  p2 = ...
else:
  ln. 46ff. 
`

**Comment by mcwimm** [comment]
Not tested, but seems plausible.

**Pull request merged**